File: mvn_java_proxy/src/main/java/mystdeim/vertx_examples/mvn_java_proxy/service/impl/AccountServiceImpl.java
package mystdeim.vertx_examples.mvn_java_proxy.service.impl;

import java.util.HashMap;
import java.util.Map;

import io.vertx.core.AsyncResult;
import io.vertx.core.Future;
import io.vertx.core.Handler;
import io.vertx.core.Vertx;
import mystdeim.vertx_examples.mvn_java_proxy.model.Account;
import mystdeim.vertx_examples.mvn_java_proxy.service.AccountService;

/**
 * @author Roman Novikov
 */
public class AccountServiceImpl implements AccountService {

    public AccountServiceImpl(Vertx vertx) {
        accounts = new HashMap<>();
    }

    Map<Long, Account> accounts;

    public void create(Account account, Handler<AsyncResult<Long>> handler) {
        accounts.put(account.getId(), account);
        handler.handle(Future.succeededFuture(account.getId()));
    }

    public void get(long id, Handler<AsyncResult<Account>> handler) {
        handler.handle(Future.succeededFuture(accounts.get(id)));
    }
}

File: mvn_kotlin_proxy/src/main/kotlin/mystdeim/vertx_examples/mvn_kotlin_proxy/service/impl/AccountServiceImpl.kt
package mystdeim.vertx_examples.mvn_kotlin_proxy.service.impl

import java.util.HashMap

import io.vertx.core.AsyncResult
import io.vertx.core.Future
import io.vertx.core.Handler
import io.vertx.core.Vertx
import mystdeim.vertx_examples.mvn_kotlin_proxy.model.Account
import mystdeim.vertx_examples.mvn_kotlin_proxy.service.AccountService

/**
 * @author Roman Novikov
 */
class AccountServiceImpl(vertx: Vertx) : AccountService {

    internal var accounts: MutableMap<Long, Account>

    init {
        accounts = HashMap()
    }

    override fun create(account: Account, handler: Handler<AsyncResult<Long>>) {
        accounts[account.id] = account
        handler.handle(Future.succeededFuture(account.id))
    }

    override fun get(id: Long, handler: Handler<AsyncResult<Account>>) {
        handler.handle(Future.succeededFuture(accounts[id]))
    }
}
